### PR TITLE
Add update API to operations

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* Add update script for query.

* Implement update API in Template.

* Add new exception for update domain not exist.

Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/script/BasicGremlinScriptLiteral.java [hunk 1]
```diff
@@ -32,4 +33,18 @@ protected void generateProperties(@NonNull List<String> scriptList, @NonNull fin
             }
         }
     }
+
+    protected String removeIdProperty(@NonNull String scriptLiteral) {
+        final String idProperty = "'id'";
```
Comment (sophiaso) [inline src/main/java/com/microsoft/spring/data/gremlin/conversion/script/BasicGremlinScriptLiteral.java]: Is the scriptLiteral of json format? So the id property can only be `'id'` but never `id`?

File: src/main/java/com/microsoft/spring/data/gremlin/exception/GremlinUpdationException.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.spring.data.gremlin.exception;
+
+import org.springframework.dao.DataAccessResourceFailureException;
+
+public class GremlinUpdationException extends DataAccessResourceFailureException {
```
Comment (sophiaso) [inline src/main/java/com/microsoft/spring/data/gremlin/exception/GremlinUpdationException.java]: Why it's named `Updation` but not `Update`?

File: src/main/java/com/microsoft/spring/data/gremlin/query/GremlinTemplate.java [hunk 3]
```diff
@@ -190,5 +187,22 @@ public void deleteAll() {
...
+        final Class<T> domainClass = (Class<T>) object.getClass();
+
+        final GremlinSource source = information.getGremlinSource();
+        final T found = this.findById(source.getId(), domainClass);
+
+        if (found == null) {
+            throw new GremlinUpdationException("cannot update the object doesn't exist");
+        }
+
+        return this.insert(object);
```
Comment (sophiaso) [inline src/main/java/com/microsoft/spring/data/gremlin/query/GremlinTemplate.java]: Is there any way we can update directly without find first?

File: src/main/java/com/microsoft/spring/data/gremlin/exception/GremlinUpdationException.java (hunk 2 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/spring/data/gremlin/exception/GremlinUpdationException.java]: following the previous naming rule insert to insertion.

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/script/BasicGremlinScriptLiteral.java (hunk 1 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/spring/data/gremlin/conversion/script/BasicGremlinScriptLiteral.java]: not json, but id is also valid here. will refine it.

File: src/main/java/com/microsoft/spring/data/gremlin/query/GremlinTemplate.java (hunk 3 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/spring/data/gremlin/query/GremlinTemplate.java]: Yes, we can do that without specific id value, but if the node not exist in graph. The sever may generate one for you. Which is out of our control. So if you try to find after update, you may got different id for one domain.

**Pull request closed without merging**